conans/client/cmd/uploader.py

```diff
@@ -501,59 +501,58 @@ def compress_files(files, symlinks, name, dest_dir, output=None):
     t1 = time.time()
     # FIXME, better write to disk sequentially and not keep tgz contents in memory
     tgz_path = os.path.join(dest_dir, name)
-    set_dirty(tgz_path)
-    with open(tgz_path, "wb") as tgz_handle:
-        # tgz_contents = BytesIO()
-        tgz = gzopen_without_timestamps(name, mode="w", fileobj=tgz_handle)
-
-        for filename, dest in sorted(symlinks.items()):
-            info = tarfile.TarInfo(name=filename)
-            info.type = tarfile.SYMTYPE
-            info.linkname = dest
-            info.size = 0  # A symlink shouldn't have size
-            tgz.addfile(tarinfo=info)
-
-        mask = ~(stat.S_IWOTH | stat.S_IWGRP)
-        i_file = 0
-        n_files = len(files)
-        last_progress = None
-        if output and n_files > 1 and not output.is_terminal:
-            output.write("[")
-        elif output and n_files > 1 and output.is_terminal:
-            progress_bar = tqdm(total=len(files), desc="Compressing %s" % name,
-                                unit="files", leave=True, dynamic_ncols=False,
-                                ascii=True, file=output)
-
-        for filename, abs_path in sorted(files.items()):
-            info = tarfile.TarInfo(name=filename)
-            info.size = os.stat(abs_path).st_size
-            info.mode = os.stat(abs_path).st_mode & mask
-            if os.path.islink(abs_path):
+    with set_dirty_context_manager(tgz_path):
+        with open(tgz_path, "wb") as tgz_handle:
+            # tgz_contents = BytesIO()
+            tgz = gzopen_without_timestamps(name, mode="w", fileobj=tgz_handle)
```

Comment:
why not use context for gzip as well?

Reply:
do you mean `with gzopen_without_timestamps(name, mode="w", fileobj=tgz_handle) as tgz` ? It doesn't support context manager.

Reply:
oh, this is unfortunate, in that case, will we lost archive in case of exception?

Reply:
unfortunate yes, but it's an exception case ;)